step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java

```diff
@@ -0,0 +1,38 @@
+package step.framework.server;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.FilterConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletResponse;
+import org.eclipse.jetty.server.Request;
+
+import java.io.IOException;
+
+public class CacheControlFilter implements Filter {
+
+    @Override
+    public void init(FilterConfig filterConfig) {}
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        if (servletRequest instanceof Request) {
+            Request request = (Request) servletRequest;
+            HttpServletResponse httpResponse = (HttpServletResponse) response;
+
+            // Check if the request is for the index HTML file which is also implicit for the root URI context (welcome file)
+            if (request.getRequestURI().endsWith("index.html") || request.getPathInContext().equals("/")) {
```

Comment:
What is the reason of the check request.getPathInContext().equals("/")? Just for my understanding

Reply:
@jeromecomte well this is what's in comment but apparently not very clear. There are actually no request to index.html directly just keeping it to be safe, the important request is to "http(s)://hostname:port" for which the path in context is "/". Jetty then automatically return the welcome page (i.e. index.html).